.github/workflows/main.yml

```diff
@@ -25,7 +25,7 @@ jobs:
       - name: Install Go
         uses: actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9 # v4.0.0
         with:
-          go-version: 1.20.x
+          go-version: "^1.20.6"
```

Comment:
How do you like
```
go-version-file: 'go.mod'
```
instead. Just found it in the `setup-go` docs.

Reply:
I don't think you can set patch versions in go.mod. I've updated to explicitly 1.20.6, because 1.20.5 fails the vulnerability check.

Reply:
Ah I see, I guess it would use the cached old version potentially, so let's do it your way.

Reply:
Exactly, when I set it to ^1.20 it used the cached 1.20.5 and the build failed because of it